spec/cli_spec.rb

```diff
@@ -127,12 +127,105 @@
         expect(stdout.lines[0]).to eq("F\n")
         expect(stdout).to include <<~ERR
           #{FAIL_APP_DIR}/dumpers/fail_app_dumper.rb:4 fail_app.invalid_status_code received unexpected status code 200 OK (expected 404)
-          #{Dir.getwd}/lib/rails_response_dumper/runner.rb:77:in `block (3 levels) in run_dumps': unexpected status code 200 OK (expected 404) (RuntimeError)
+          #{Dir.getwd}/lib/rails_response_dumper/runner.rb:91:in `block (3 levels) in run_dumps': unexpected status code 200 OK (expected 404) (RuntimeError)
         ERR
         expect(stdout).not_to include(invalid_number_of_statuses)
         expect(stdout).not_to include(dumper_2_invalid_status_code)
         expect(status.exitstatus).to eq(1)
       end
     end
   end
+
+  context 'with filename argument' do
+    context 'without previous dumps' do
+      context 'with one file specified' do
+        it 'only runs dumps from that file' do
+          cmd = %W[bundle exec rails-response-dumper --dumps-dir #{tmpdir} dumpers/tests_response_dumper.rb]
+          stdout, stderr, status = Open3.capture3(*cmd, chdir: APP_DIR)
+
+          expect(tmpdir).to match_snapshots(File.join(APP_DIR, 'snapshots_single_file'))
+
+          expect(stderr).to eq('')
+          expect(stdout).to eq("..\n")
+          expect(status.exitstatus).to eq(0)
+        end
+      end
+
+      context 'with two files specified' do
+        it 'runs dumps from both files' do
+          cmd = %W[
+            bundle exec rails-response-dumper --dumps-dir #{tmpdir}
+            dumpers/tests_response_dumper.rb dumpers/root_response_dumper.rb
+          ]
+          stdout, stderr, status = Open3.capture3(*cmd, chdir: APP_DIR)
+
+          expect(tmpdir).to match_snapshots(File.join(APP_DIR, 'snapshots_two_files'))
+
+          expect(stderr).to eq('')
+          expect(stdout).to eq("...\n")
+          expect(status.exitstatus).to eq(0)
+        end
+      end
+    end
+
+    context 'with previous dumps' do
+      before do
+        cmd = %W[bundle exec rails-response-dumper --dumps-dir #{tmpdir}]
+        Open3.capture3(*cmd, chdir: APP_DIR)
+      end
+
+      context 'with no changes to the dump file' do
+        it 'does not alter the existing dumps from any file' do
+          cmd = %W[bundle exec rails-response-dumper --dumps-dir #{tmpdir} dumpers/tests_response_dumper.rb]
+          stdout, stderr, status = Open3.capture3(*cmd, chdir: APP_DIR)
+
+          expect(tmpdir).to match_snapshots(File.join(APP_DIR, 'snapshots'))
+
+          expect(stderr).to eq('')
+          expect(stdout).to eq("..\n")
+          expect(status.exitstatus).to eq(0)
+        end
+      end
+
+      context 'with changes to the dump file' do
+        before do
+          FileUtils.mv(
+            "#{APP_DIR}/dumpers/tests_response_dumper.rb",
+            "#{APP_DIR}/configurable_dumpers/initial_tests_response_dumper.rb"
+          )
```

Comment:
What is going on here? I don't think we should be modifying the repo files directly. If we need to changes files mid-test, we should be working out of a temporary directory.

Reply:
A couple things occurred to me when going to do this:

1) This is the PR in which we make the dumper directory configurable, so no need for an early PR 🤦 
2) We don't need to actually move the files to test the new file content, we can just reference the altered version of the file directly from configurable_dumps.

Let me know what you think: [Reference configurable directory directly](https://github.com/Pioneer-Valley-Books/rails-response-dumper/pull/63/commits/61212e1b803c8fa8acafe5c634eaa2602a868f88)